tools/flash_shelly.py

```diff
@@ -0,0 +1,501 @@
+#!/usr/bin/env python3
+#  Copyright (c) 2020 Andrew Blackburn & Deomid "rojer" Ryabkov
+#  All rights reserved
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#  This script will probe for any shelly device on the network and it will
+#  attempt to update them to the latest firmware version available.
+#  This script will not flash any firmware to a device that is not already on a
+#  version of this firmware, if you are looking to flash your device from stock
+#  or any other firmware please follow instructions here:
+#  https://github.com/mongoose-os-apps/shelly-homekit/wiki
+#
+#  Shelly HomeKit flashing script utility
+#  Usage: -{m|l|a|n|y|h} {hostname(s) optional}
+#   -m {homekit|revert|keep}   Script mode.
+#   -l                         List info of shelly device.
+#   -a                         Run against all the devices on the network.
+#   -n                         Do a dummy run through.
+#   -y                         Do not ask any confirmation to perform the flash.
+#   -V                         Force a particular version.
+#   -D                         Enable Debug Logging, you can increase logging with (-DD).")
+#   -h                         This help text.
+#
+#  usage: python3 flash_shelly.py -la
+#  usage: python3 flash_shelly.py shelly1-034FFF
+
+import functools
+import getopt
+import importlib
+import importlib.util
+import json
+import logging
+import os
+import platform
+import re
+import subprocess
+import sys
+import time
+import urllib
+
+logging.TRACE = 5
+logging.addLevelName(logging.TRACE, 'TRACE')
+logging.Logger.trace = functools.partialmethod(logging.Logger.log, logging.TRACE)
+logging.trace = functools.partial(logging.log, logging.TRACE)
+logging.basicConfig(format='%(message)s')
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+arch = platform.system()
+# Windows does not support acsii colours
+WHITE = '\033[1m' if not arch.lower().startswith('win') else ""
```

Comment:
actually, this depends on terminal capabilities, not OS. maybe there's a way to figure it out.
but in any case, - avoid repetition where possible:

```
if not arch.lower().startswith('win'):
  ...
else:
  WHITE = RED = ... = NC = ""
```

Reply:
doh ye np, don't know what I was thinking there.

Reply:
https://github.com/mongoose-os-apps/shelly-homekit/pull/176/commits/758d4c4daeee3692a62120ec21df1d122c2d1736